pkg/snapshot/restorer/restorer_test.go

```diff
@@ -961,62 +1009,63 @@ func corruptEtcdDir() error {
 
 //takeValidV1Snaps saves valid snaps in the v1 prefix dir of snapstore so that restorer could restore from them
 //TODO: Consider removing when backward compatibility no longer needed
-func takeValidV1Snaps(logger *logrus.Entry, container string, deltaSnapshotPeriod time.Duration, endpoints []string, stopCh <-chan struct{}, compressionConfig *compressor.CompressionConfig, mode string) error {
+func takeValidV1Snaps(logger *logrus.Entry, container string, resp *utils.EtcdDataPopulationResponse, deltaSnapshotPeriod time.Duration, endpoints []string, stopCh <-chan struct{}, compressionConfig *compressor.CompressionConfig, mode string) error {
 	//Here we run the snapshotter to take snapshots. The snapshotter by default stores the snaps in the v2 directory.
```

Comment:
Can you encapsulate this functionality into a separate function like `takeV2Snaps` and then use it within `takeValidV1Snaps` and `takeInvalidV1Snaps` functions? So that you can also reuse the `takeV2Snaps` function in all the test cases under `Running Restorer when both v1 and v2 directory structures are present` too.

Reply:
Hmm.. I need to retouch these test cases again.

Reply:
Done